File: SecondYear/AvantikaSharma2307_AvantikaSharma_2226cseai1008_2/Week_1/pattern9.py
#User function Template for python3

class Solution:
    def printTriangle(self, N):
        for l in range(1,N+1,1):
            for a in range(1,N+1,1):
                if a<=(N+1-l):
                    print("*",end="")
                else:
                    print(" ",end="")
            for b in range(1,N+1,1):
                if b<l:
                    print(" ",end="")
                else:
                    print("*",end="")
                
            print()      
        for i in range(1,N+1,1):
            for j in range(1,N+1,1):
                if j<=i:
                    print("*",end="")
                else:
                    print(" ",end="")
            for k in range(1,N+1,1):
                if k>=(N+1-i):
                    print("*",end="")
                else:
                    print(" ",end="")
            print()    
           

#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        N = int(input().strip())
        ob = Solution()
        ob.printTriangle(N)
# } Driver Code Ends

File: SecondYear/seeeyaa_PariGupta_2226CSE(AI)1096_2/Week_1/Pattern9.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>

using namespace std;

// } Driver Code Ends
class Solution {
  public:
    void printTriangle(int n)
    {
        int i,j;
        for(i=0;i<n;i++)
        {
            for(j=1;j<=n-i;j++)
            {
                cout<<'*';
            }
            for(j=2*n;j>2*(n-i);j--)
            {
                cout<<" ";
            }
             for(j=1;j<=n-i;j++)
            {
                cout<<'*';
            }
            cout<<endl;
        }
        for(i=1;i<=n;i++)
        {
            for(j=1;j<=i;j++)
            {
                cout<<'*';
            }
            for(j=2*n;j>2*i;j--)
            {
                cout<<" ";
            }
             for(j=1;j<=i;j++)
            {
                cout<<'*';
            }
            cout<<endl;
        }
    }
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;

        Solution ob;
        ob.printTriangle(n);
    }
    return 0;
}
// } Driver Code Ends